runtime-modules/content-directory/src/lib.rs

```diff
@@ -929,36 +929,24 @@ decl_module! {
 
             Self::ensure_all_properties_are_valid(&new_properties)?;
 
-            let class_properties = class.get_properties_ref();
+            let class_properties = class.get_properties();
 
-            Self::ensure_all_property_names_are_unique(class_properties, &new_properties)?;
+            Self::ensure_all_property_names_are_unique(&class_properties, &new_properties)?;
 
-            // Create new Schema with existing properies provided
-            let mut schema = Schema::new(existing_properties);
+            Self::ensure_schema_properties_are_valid_indices(&existing_properties, &class_properties)?;
 
-            schema.ensure_schema_properties_are_valid_indices(class_properties)?;
+            let schema = Self::create_class_schema(existing_properties, &class_properties, &new_properties);
```

Comment:
I believe everything from `let schema`can be placed under the mutation comment, because the prior check is the last one needed to ensure mutation safety.

Reply:
https://github.com/iorveth/substrate-runtime-joystream/commit/37eb8480ad4cfc6368bf77bd0a8d3624b3761b89